core/src/test/java/org/apache/iceberg/TestTransaction.java

```diff
@@ -130,12 +133,14 @@ public void testMultipleOperationTransaction() {
         statuses(Status.DELETED, Status.EXISTING));
 
     Assert.assertEquals(
-        "Table should have a snapshot for each operation", 2, readMetadata().snapshots().size());
+        "Table should have a snapshot for each operation", 3, readMetadata().snapshots().size());
     validateManifestEntries(
-        readMetadata().snapshots().get(0).allManifests(table.io()).get(0),
+        readMetadata().snapshots().get(1).allManifests(table.io()).get(0),
         ids(appendSnapshot.snapshotId(), appendSnapshot.snapshotId()),
         files(FILE_A, FILE_B),
         statuses(Status.ADDED, Status.ADDED));
+
+    org.assertj.core.api.Assertions.assertThat(table.history()).containsAll(initialHistory);
```

Comment:
You might consider asserting on the entire history that should be there -- which should be the initial history + the one entry post `txn.commitTransaction()`.